teepod/src/main_service.rs

```diff
@@ -241,6 +241,48 @@ impl TeepodRpc for RpcHandler {
             public_key: response.public_key,
         })
     }
+
+    async fn get_info(self, request: Id) -> Result<GetInfoResponse> {
+        if let Some(vm) = self.app.get_vm(&request.id) {
+            Ok(GetInfoResponse {
+                found: true,
+                info: Some(vm),
+            })
+        } else {
+            Ok(GetInfoResponse {
+                found: false,
+                info: None,
+            })
+        }
+    }
+
+    async fn resize_vm(self, request: ResizeVmRequest) -> Result<()> {
+        let vm = self.app
+            .get_vm(&request.id)
+            .ok_or_else(|| anyhow::anyhow!("vm not found: {}", request.id))?;
+        if vm.status != "stopped" {
+            return Err(anyhow::anyhow!("vm should be stopped before resize: {}", request.id));
+        }
+        let work_dir = self.app.config.run_path.join(&request.id);
+        let vm_work_dir = VmWorkDir::new(&work_dir);
+        let mut manifest = vm_work_dir
+            .manifest()
+            .context("failed to read manifest")?;
+        if let Some(vcpu) = request.vcpu {
+            manifest.vcpu = vcpu;
+        }
+        if let Some(memory) = request.memory {
+            manifest.memory = memory;
+        }
+        if let Some(disk_size) = request.disk_size {
+            manifest.disk_size = disk_size;
```

Comment:
Note: Resizing the disk size will not take effect

Reply:
Let me add a comment regarding this situation and it's ok we only updating the manifest here for now.